Adding get started topics (#471)
osisoft/OCS-Docs: content/add-organize-data/organize-data/assets/asset-concept.md

---
uid: ccAssets
---
# Assets

> An asset is a container that groups information about an asset. An asset is composed of the metadata or static information about the asset, such as location or company, and properties which are references to streams in the Sequential Data Store. Assets allow users to retrieve all of the stream values associated with an asset and to provide context to this dynamic data. 

Reviewer: Wow, that first sentence is something else. “An asset is a container that groups information about an asset.”  I don't know how to define it otherwise, but it sure seems like you're defining a term by using the term. And that's confusing.

> An asset is a container that groups information about an asset. An asset is composed of the metadata or static information about the asset, such as location or company, and properties which are references to streams in the Sequential Data Store. Assets allow users to retrieve all of the stream values associated with an asset and to provide context to this dynamic data. 

Author: ```suggestion
 An asset is a container for metadata or static information about the asset (such as location or company) and properties which are references to streams in the Sequential Data Store. Assets allow users to retrieve all of the stream values associated with an asset and to provide context to this dynamic data. 
```

---
uid: ccAssets
---
# Assets

An asset is a container that groups information about an asset. An asset is composed of the metadata or static information about the asset, such as location or company, and properties which are references to streams in the Sequential Data Store. Assets allow users to retrieve all of the stream values associated with an asset and to provide context to this dynamic data. 

## PI Server counterpart

Assets are comparable to elements in PI AF server. Similar to elements, OCS assets include both dynamic and static information associated with the asset. The static information provides context for the dynamic data. Assets in OCS are exposed as a flat list that can be searched using metadata values. This contrasts with the elements in PI AF server which are organized in a hierarchy.

## Assets best practices

The following best practices are recommended when you create your assets:

- Use OCS assets to bring together in one place the data from multiple streams and the static data associated with a logical asset. OCS assets make it easier to visualize and contextualize the data for a given logical asset.

> - Where possible, define the unit of measure (UOM) in the Sequential Data Store types rather than defining it in the asset. This ensures that the UOM is known even when the stream is directly accessed. <!-- What does this mean? Will users know what this means? -->

Reviewer 2: I might approach it a different way since the focus here is on Assets best practices, Really what we want to convey and say here is the best practice is not to do this and have this information already on the Type or Stream.d   

Note: While a UoM can be defined on the asset property, for consistency across data retrieval endpoints for the underlying data you should define the UoM on the Type or Stream.

> - Where possible, define the unit of measure (UOM) in the Sequential Data Store types rather than defining it in the asset. This ensures that the UOM is known even when the stream is directly accessed. <!-- What does this mean? Will users know what this means? -->

Author: ```suggestion
- Where possible, define the unit of measure (UOM) in the Sequential Data Store types rather than configuring the UOM on the asset.  Note: Any changes that are made to the UOM configuration on the asset are saved on the stream.  
```
<!-- I don't want to mention here that if they click Save on the UOM dialog, the change is saved to the stream even if they delete the asset or they back out of the changes on the Properties tab. They need the context of the UI for this to make sense to them. -->

> - Where possible, define the unit of measure (UOM) in the Sequential Data Store types rather than defining it in the asset. This ensures that the UOM is known even when the stream is directly accessed. <!-- What does this mean? Will users know what this means? -->

Reviewer 3: @lasato-OSI I like that suggestion but I might change to `configuring the UOM *from* the asset` (since it isn't actually saved on the asset) and `are saved *to* the stream`.

> - Where possible, define the unit of measure (UOM) in the Sequential Data Store types rather than defining it in the asset. This ensures that the UOM is known even when the stream is directly accessed. <!-- What does this mean? Will users know what this means? -->

Author: Okay, I made the change.

> - Where possible, define the unit of measure (UOM) in the Sequential Data Store types rather than defining it in the asset. This ensures that the UOM is known even when the stream is directly accessed. <!-- What does this mean? Will users know what this means? -->

Reviewer 3: Agreed, looks like we had a few threads about the same thing.

> - <!-- Doug said --> Defining a different UOM in the asset does not perform any type of conversion from UOMs defined on the asset type or SDS type, so if the UOM is defined in multiple places, they should always match.

Author: ```suggestion
- <!-- Can we say this? --> Whenever possible, OSIsoft recommends that you define the UOM on the type. The UOM can also be defined on the stream or asset, but if it is defined in multiple places, ensure that the UOMs match. If a different UOM is configured, there is no conversion between the UOMs.
```
Which UOM definition takes precedence? If it's defined on the type, then I assume that takes precedence over anything else. If it's not defined on the type and it's defined on the stream, that takes precedence over any configuration on the asset. If that correct? Is this obvious to users and, if not, do we need to be explicit about this?

Reviewer 3: Your suggestion sounds fine to me. The definition on the stream (stream property *override*) takes precedence over the definition on the type. There is no actual configuration on the asset, if you configure from the assets view it actually just sets the stream property override. 
I doubt it's all obvious to users, although I do think at least the uom configuration on the assets page is pretty explicit about affecting the streams directly instead of the asset.

Author: I'm getting confused with these different conversations. I think we want to abandon this in favor of the suggestion below this one.

> - <!-- Is this what you mean? --> Specify the UOM that is defined on the stream.  If you define a different UOM on the asset than is defined in the SDS type, there is no data conversion from one UOM to the other. <!-- Does this mean that the value remains the same. For example, if the UOM was "feet" and the value was 12, if you change the UOM to "yards," it will appear as 12 yards, not 4 yards. -->

Reviewer 3: Suggest: `Configuring a stream to use a different UOM from the asset does not perform any type of conversion if there is already a UOM defined on the SDS type itself.`

> - <!-- Is this what you mean? --> Specify the UOM that is defined on the stream.  If you define a different UOM on the asset than is defined in the SDS type, there is no data conversion from one UOM to the other. <!-- Does this mean that the value remains the same. For example, if the UOM was "feet" and the value was 12, if you change the UOM to "yards," it will appear as 12 yards, not 4 yards. -->

Author: I think we can abandon this in favor of the paragraph above: 
Whenever possible, OSIsoft recommends that you define the UOM on the type. The UOM can also be defined on the stream or asset, but if it is defined in multiple places, ensure that the UOMs match. If a different UOM is configured, there is no conversion between the UOMs.

> - <!-- Is this what you mean? --> Specify the UOM that is defined on the stream.  If you define a different UOM on the asset than is defined in the SDS type, there is no data conversion from one UOM to the other. <!-- Does this mean that the value remains the same. For example, if the UOM was "feet" and the value was 12, if you change the UOM to "yards," it will appear as 12 yards, not 4 yards. -->

Reviewer 3: That sounds fine, except I would remove `or asset`. Configuring UOM from the assets page actually sets it on the stream, not the asset - you can only configure it on the type or the stream.
